pinot-minion/src/main/java/com/linkedin/pinot/minion/taskfactory/TaskFactoryRegistry.java

```diff
@@ -42,8 +42,7 @@
 public class TaskFactoryRegistry {
   private final Map<String, TaskFactory> _taskFactoryRegistry = new HashMap<>();
 
-  public TaskFactoryRegistry(@Nonnull TaskExecutorRegistry taskExecutorRegistry,
-      @Nonnull final MinionContext minionContext) {
+  public TaskFactoryRegistry(@Nonnull TaskExecutorRegistry taskExecutorRegistry) {
```

Comment:
did we discuss changing the interface now? to add a flag to allow us to not use table config as source of truth? or was that later?

Reply:
We still want to use table config right?

Reply:
For now, yes. I thought we wanted to add the option to use other methods so that we can migrate in the future/allow tasks to not have to set task config. But that can be in another pr.

Reply:
we will use table config.